### PR TITLE
ROA Overlapping INVALID ROA and VALID ROA fix

### PR DESCRIPTION
**Pull request by MiguelAHM**
Fix the situation of both ROAS overlapping the Route
If one of the roas is VALID we should return VALID even if the other if INVALID

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java [hunk 1]
```diff
@@ -57,7 +57,8 @@ private RpslMessage validateRoa(final RpslObject rpslObject){
         return switch (invalidRpkiRoa.getValue()) {
             case INVALID_ORIGIN -> new RpslMessage(QueryMessages.roaRouteOriginConflicts(rpslObject.getType().getName(), invalidRpkiRoa.getKey().getMaxLength(), invalidRpkiRoa.getKey().getAsn()));
             case INVALID_PREFIX_LENGTH -> new RpslMessage(QueryMessages.roaRoutePrefixLengthConflicts(rpslObject.getType().getName(), invalidRpkiRoa.getKey().getMaxLength(), invalidRpkiRoa.getKey().getAsn()));
-            default -> new RpslMessage(QueryMessages.roaRouteConflicts(rpslObject.getType().getName(), invalidRpkiRoa.getKey().getMaxLength(), invalidRpkiRoa.getKey().getAsn()));
+            case INVALID_PREFIX_AND_ORIGIN -> new RpslMessage(QueryMessages.roaRouteConflicts(rpslObject.getType().getName(), invalidRpkiRoa.getKey().getMaxLength(), invalidRpkiRoa.getKey().getAsn()));
+            default -> null;
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java:61]: Add `Nullable` annotation if method returns null

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java [hunk 2]
```diff
@@ -11,10 +11,14 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
 
 import static net.ripe.db.whois.common.rpki.ValidationStatus.INVALID;
 import static net.ripe.db.whois.common.rpki.ValidationStatus.INVALID_ORIGIN;
+import static net.ripe.db.whois.common.rpki.ValidationStatus.INVALID_PREFIX_AND_ORIGIN;
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java:19]: Document that `WhoisRoaChecker` validates *overlaps*. That makes it different from other `RpkiRoaChecker` implementations.

Also consider renaming `RpkiRoaChecker` as it's confusing with `WhoisRoaChecker` (i.e. which one is abstract?)

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java [hunk 3]
```diff
@@ -2007,6 +2007,27 @@ public void search_route_roa_origin_and_prefix_length_mismatch_validation_enable
                 "either remove this route: object or update or delete the ROA.\n"));
     }
 
+    @Test
+    public void search_route_one_invalid_one_valid_roas_makes_it_valid() {
+        databaseHelper.addObject(RpslObject.parse("" +
+                "route:           92.38.45.0/24\n" +
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:2013]: Where is the valid ROA?

I can only find the ROA with prefix 92.142.0.0/16 and ASN AS3215 in roas.json.

Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:2013]: is exactly same example that Ricardo gave to us
I did debug and I have two 

    { "asn": "AS44546", "prefix": "92.38.0.0/17", "maxLength": 24, "ta": "arin" },
    { "asn": "AS61979", "prefix": "92.38.44.0/22", "maxLength": 24, "ta": "ripe" }

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java:19]: well WhoisRoaChecker is extending RpkiRoaChecker so the abstract class is RpkiRoaChecker
I can add "Abstract" to make it more clear

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java (hunk 3 above)
Comment (eshryane) [reply whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:2013]: OK, I see the extra ROAs in the branch (diff)

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java (hunk 2 above)
Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java:18]: Thanks

**Review by eshryane — APPROVED** [review]

**Pull request merged**